ansible/verify_config.py

```diff
@@ -0,0 +1,466 @@
+                    f"conf-name='{testbed['conf-name']}'. Previously declared as {attribute}="
+                    f"{group_name_attributes[attribute]} in conf-name={group_name_attributes['conf-name']}")
+                is_valid = False
+
+        return is_valid
+
+    def _server_name_must_be_in_veos_file(self, testbed) -> bool:
+        try:
+            return (testbed['server'] in
+                    Utility.parse_yml(Config.VEOS_FILE.value)['all']['children']['servers']['children'])
+        except KeyError as unknown_key:
+            self.assertion.add_error_details(f"Key not found: {unknown_key}")
+            self.assertion.log_error(
+                "veos file is not in the correct format. Update the file or update this script",
+                error_file=Config.VEOS_FILE.value,
+                error_type="error"
+            )
+
+    def _topo_name_must_be_in_veos_file(self, testbed) -> bool:
+        try:
+            return testbed['topo'] in Utility.parse_yml(Config.VEOS_FILE.value)['all']['children']['servers']['vars'][
+                'topologies']
+        except KeyError as unknown_key:
+            self.assertion.add_error_details(f"Key not found: {unknown_key}")
+            self.assertion.log_error(
+                "veos file is not in the correct format. Update the file or update this script",
+                error_file=Config.VEOS_FILE.value,
+                error_type="error"
+            )
+
+    def _required_attributes_must_be_in_testbed(self, testbed: Dict['str', 'str']) -> bool:
+        is_valid = True
+        required_attributes = {
+            "conf-name",
+            "group-name",
+            "topo",
+            "ptf_image_name",
+            "ptf_ip",
+            "server",
+            "vm_base",
+            "dut",
+            "inv_name",
+            "auto_recover",
+            "comment"
+        }
+
+        missing_keys = required_attributes - testbed.keys()
+
+        if missing_keys:
+            self.assertion.add_error_details(f"Found missing required keys: {missing_keys}")
+            is_valid = False
+
+        return is_valid
+
+
+class InventoryNameValidator(Validator):
+    def __init__(self):
+        super().__init__(Config.FANOUT_GRAPH_GROUP_FILE.value)
+
+    def validate(self):
+        self.assertion.assert_true(lambda: self._inv_name_from_devices_files_must_be_the_same_as_graph_group_yml_file(),
+                                   reason="Inventory name must be consistence between "
+                                          f"{Formatting.bold(Config.FANOUT_GRAPH_GROUP_FILE.value)} and "
+                                          f"{Formatting.bold(Config.FANOUT_DEVICES_FILE.value)}")
+        self.assertion.assert_true(lambda: self._check_if_inv_name_has_inv_file(),
+                                   reason="Inventory should have an inventory file")
+
+    def _inv_name_from_devices_files_must_be_the_same_as_graph_group_yml_file(self):
```

Comment:
Consider renaming this to `validate_inv_name_consistency()` to make it more concise and readable